plugins/gateway-backend/src/router.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { DiscoveryService, LoggerService } from '@backstage/backend-plugin-api';
+import { InstanceMetadataService } from '@backstage/backend-plugin-api/alpha';
+import { Request, Response, NextFunction } from 'express';
+import { createProxyMiddleware } from 'http-proxy-middleware';
+import { context } from '@opentelemetry/api';
+import { getRPCMetadata } from '@opentelemetry/core';
+
+export function createRouter({
+  discovery,
+  instanceMeta,
+}: {
+  discovery: DiscoveryService;
+  instanceMeta: InstanceMetadataService;
+  logger: LoggerService;
+}) {
+  const localPluginIds = new Set(
+    instanceMeta
+      .getInstalledFeatures()
+      .filter(f => f.type === 'plugin')
+      .map(f => f.pluginId),
+  );
+
+  const proxy = createProxyMiddleware({
```

Comment:
Wonder if it might be worth bringing in the fix from https://github.com/backstage/backstage/pull/29498 here too?

Reply:
mmm [this test](https://github.com/backstage/backstage/pull/29498/files#diff-282f493556b02275f0ae86b16dc6b3f40b6e970673ed142fad5d23f13493930bR609) doesn't seem to cover the entire e2e flow. I will try to see if I can quickly test it properly

Reply:
alright added https://github.com/backstage/backstage/pull/29329/commits/e6947e1790f540fc2f7e6b63a530fab2b687205a 🚀